helix-term/src/ui/editor.rs

```diff
@@ -154,6 +120,25 @@ impl EditorView {
             highlights = Box::new(syntax::merge(highlights, diagnostic));
         }
 
+        // Set DAP highlights, with possibility to overwrite syntax highlights, if fg set.
+        let highlights: Box<dyn Iterator<Item = HighlightEvent>> =
+            if let Some(frame) = editor.current_stack_frame() {
+                let dap_line = frame.line - 1;
+                if let Some(dap_current_index) = theme.find_scope_index("ui.highlight.frameline") {
+                    let text = doc.text();
+                    let dap_line_start = text.line_to_char(dap_line);
+                    let dap_line_end = text.line_to_char(dap_line + 1);
+                    Box::new(syntax::merge(
+                        highlights,
```

Comment:
No that we only highlight the background wouldn't it make more sense to highlight the whole line instead of just the text. Otherwise an empty line might endup not being invisible (when compiling with optimizations a breakpoint sometimes endsup on an empty line). Inoeraonally prefer the look too. If you don't want to draw over the gutter that's possible that way too

Reply:
I am not too confident in highlighting the whole line, since it would be a behaviour very different from how almost all IDEs work in this case. For the empty line we still have the indicator as well and a highlight if you also have the EoL character enabled. I also think it makes a bit more sense to have it as is since it clearly tells you we are at this point _in the code_, rather than you are at line 3 in you editor which happens to have this code in it.

Ideally, a followup to this PR would make the highlight scope based as well, think debugging a lambda, where only part of the code line would be highlighted, to reflect where _in the code_ execution has stopped at.

Let me know your thoughts on it, but I do think the current approach is a bit more "forgiving" with users and guides their attention in a helpful way.

Reply:
I'm split on this too, I see some editors use a whole line highlight, but highlighting a section of the code might be useful in case you're stepping through something like `foo(bar(), baz())` -- you can distinguish what parameter it stopped on

Reply:
Exactly what I was thinking about, or for  `.map(|a_var| if var.field { ... })`

Reply:
That is something I don't know, but DAP supports column level precision for breakpoints, so there might be a way

Reply:
I think I now understand what you mean, @pascalkuthe, after taking another look. I agree, it should look as this:

![image](https://user-images.githubusercontent.com/46979062/224118568-1b7f1c69-82e8-4214-8dee-11611c832ed8.png)

Reply:
If the DAP supports column-level precision for breakpoints then it should be doable. The `stopped` event returns `hitBreakpointIds` which can help determine which breakpoint on that line was hit but we currently don't use that param

Reply:
After a bit more talks, the current implementation, meaning not the whole line is highlighted might be the better approach for now, since it allows scope highlighting further down the line (#6238). Once @pascalkuthe finishes his highlighting improvements we could make it possible to achieve this UI as well

Reply:
Let's use the whole line for now since that matches current `master` behaviour. We can improve once it's possible to highlight just a part of the line

Reply:
ok I will use that behaviour instead and add this more detailed control in #6238. are the current colours to your liking as well? or do you want me to experiment a bit more @archseer?